node-api/backend/block.go

```diff
@@ -29,7 +29,7 @@ import (
 )
 
 // BlockHeader returns the block header at the given slot.
-func (b Backend) BlockHeaderAtSlot(slot math.Slot) (*ctypes.BeaconBlockHeader, error) {
+func (b *Backend) BlockHeaderAtSlot(slot math.Slot) (*ctypes.BeaconBlockHeader, error) {
 	st, _, err := b.stateFromSlot(slot)
```

Comment:
Had to make the pointer receiver as we have mutex in Backend. Same goes for other places in the PR.

Reply:
We don't have mutex anymore, still keeping the pointer changes. Will ensure it is there for every endpoint as part of a separate PR.